src/game/logic/scriptengine/scriptengine.cpp

```diff
@@ -13,56 +13,9401 @@
+    m_actionTemplates[ScriptAction::NAMED_SET_SPECIAL_POWER_COUNTDOWN].m_uiName =
+        "Unit_/ Special power countdown timer -- set.";
+    m_actionTemplates[ScriptAction::NAMED_SET_SPECIAL_POWER_COUNTDOWN].m_numParameters = 3;
+    m_actionTemplates[ScriptAction::NAMED_SET_SPECIAL_POWER_COUNTDOWN].m_parameters[0] = Parameter::UNIT;
+    m_actionTemplates[ScriptAction::NAMED_SET_SPECIAL_POWER_COUNTDOWN].m_parameters[1] = Parameter::SPECIAL_POWER;
+    m_actionTemplates[ScriptAction::NAMED_SET_SPECIAL_POWER_COUNTDOWN].m_parameters[2] = Parameter::INT;
+    m_actionTemplates[ScriptAction::NAMED_SET_SPECIAL_POWER_COUNTDOWN].m_numUiStrings = 4;
+    m_actionTemplates[ScriptAction::NAMED_SET_SPECIAL_POWER_COUNTDOWN].m_uiStrings[0] = "Set ";
+    m_actionTemplates[ScriptAction::NAMED_SET_SPECIAL_POWER_COUNTDOWN].m_uiStrings[1] = "'s ";
+    m_actionTemplates[ScriptAction::NAMED_SET_SPECIAL_POWER_COUNTDOWN].m_uiStrings[2] = " to ";
+    m_actionTemplates[ScriptAction::NAMED_SET_SPECIAL_POWER_COUNTDOWN].m_uiStrings[3] = " seconds.";
+
+    m_actionTemplates[ScriptAction::NAMED_ADD_SPECIAL_POWER_COUNTDOWN].m_internalName = "NAMED_ADD_SPECIAL_POWER_COUNTDOWN";
+    m_actionTemplates[ScriptAction::NAMED_ADD_SPECIAL_POWER_COUNTDOWN].m_uiName =
+        "Unit_/ Special power countdown timer -- add seconds.";
+    m_actionTemplates[ScriptAction::NAMED_ADD_SPECIAL_POWER_COUNTDOWN].m_numParameters = 3;
+    m_actionTemplates[ScriptAction::NAMED_ADD_SPECIAL_POWER_COUNTDOWN].m_parameters[0] = Parameter::UNIT;
+    m_actionTemplates[ScriptAction::NAMED_ADD_SPECIAL_POWER_COUNTDOWN].m_parameters[1] = Parameter::SPECIAL_POWER;
+    m_actionTemplates[ScriptAction::NAMED_ADD_SPECIAL_POWER_COUNTDOWN].m_parameters[2] = Parameter::INT;
+    m_actionTemplates[ScriptAction::NAMED_ADD_SPECIAL_POWER_COUNTDOWN].m_numUiStrings = 4;
+    m_actionTemplates[ScriptAction::NAMED_ADD_SPECIAL_POWER_COUNTDOWN].m_uiStrings[0] = " ";
+    m_actionTemplates[ScriptAction::NAMED_ADD_SPECIAL_POWER_COUNTDOWN].m_uiStrings[1] = "'s ";
+    m_actionTemplates[ScriptAction::NAMED_ADD_SPECIAL_POWER_COUNTDOWN].m_uiStrings[2] = " has ";
+    m_actionTemplates[ScriptAction::NAMED_ADD_SPECIAL_POWER_COUNTDOWN].m_uiStrings[3] = " seconds added to it.";
+
+    m_actionTemplates[ScriptAction::SKIRMISH_FIRE_SPECIAL_POWER_AT_MOST_COST].m_internalName =
+        "SKIRMISH_FIRE_SPECIAL_POWER_AT_MOST_COST";
+    m_actionTemplates[ScriptAction::SKIRMISH_FIRE_SPECIAL_POWER_AT_MOST_COST].m_uiName =
+        "Skirmish_/ Special power -- fire at enemy's highest cost area.";
+    m_actionTemplates[ScriptAction::SKIRMISH_FIRE_SPECIAL_POWER_AT_MOST_COST].m_numParameters = 2;
+    m_actionTemplates[ScriptAction::SKIRMISH_FIRE_SPECIAL_POWER_AT_MOST_COST].m_parameters[0] = Parameter::SIDE;
+    m_actionTemplates[ScriptAction::SKIRMISH_FIRE_SPECIAL_POWER_AT_MOST_COST].m_parameters[1] = Parameter::SPECIAL_POWER;
+    m_actionTemplates[ScriptAction::SKIRMISH_FIRE_SPECIAL_POWER_AT_MOST_COST].m_numUiStrings = 2;
```

Comment:
number of ui strings does not match set strings.

Reply:
Original code does this.